src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -0,0 +1,348 @@
+        [LocalizedName(typeof(CSharpVSResources), nameof(CSharpVSResources.Sort_usings))]
+        public static readonly FixIdDefinition SortImports;
+
+        public override Task<bool> FixAsync(ICodeCleanUpScope scope, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
+        {
+            switch (scope)
+            {
+                case TextBufferCodeCleanUpScope textBufferScope:
+                    return FixTextBufferAsync(textBufferScope, context, cancellationToken);
+                case IVsHierarchyCodeCleanupScope hierarchyContentScope:
+                    return FixHierarchyContentAsync(hierarchyContentScope, context, cancellationToken);
+                default:
+                    return Task.FromResult(false);
+            }
+        }
+
+        private Task<bool> FixHierarchyContentAsync(IVsHierarchyCodeCleanupScope hierarchyContent, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
+        {
+            // TODO: this one will be implemented later
+            // https://github.com/dotnet/roslyn/issues/30165
+            var hierarchy = hierarchyContent.Hierarchy;
+            if (hierarchy == null)
+            {
+                // solution
+                return Task.FromResult(false);
+            }
+
+            var itemId = hierarchyContent.ItemId;
+
+            if (hierarchy.GetCanonicalName(itemId, out var path) == 0)
+            {
+                var attr = File.GetAttributes(path);
+                if (attr.HasFlag(FileAttributes.Directory))
+                {
+                    // directory
+                }
+                else
+                {
+                    // document
+                }
+            }
+
+            return Task.FromResult(false);
+        }
+
+        private async Task<bool> FixTextBufferAsync(TextBufferCodeCleanUpScope textBufferScope, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
+        {
+            using (var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(context.OperationContext.UserCancellationToken, cancellationToken))
+            {
+                cancellationToken = cancellationTokenSource.Token;
+
+                var buffer = textBufferScope.SubjectBuffer;
+                if (buffer != null)
+                {
+                    using (var scope = context.OperationContext.AddScope(allowCancellation: true, description: EditorFeaturesResources.Applying_changes))
+                    {
+                        var progressTracker = new ProgressTracker((description, completed, total) =>
+                        {
+                            if (scope != null)
+                            {
+                                scope.Description = description;
+                                scope.Progress.Report(new ProgressInfo(completed, total));
+                            }
+                        });
```

Comment:
i'd love a helper that gave you one of these.  But, for now, this seems fine to me :)